provision/kubernetes/nodecontainer.go

```diff
@@ -93,6 +93,13 @@ func (m *nodeContainerManager) deployNodeContainerForCluster(client *ClusterClie
 			},
 		},
 	}
+	singlePool, err := client.SinglePool()
+	if err != nil {
+		return err
+	}
+	if singlePool && pool != "" {
```

Comment:
why we need to evaluate `pool != ""` ?

Reply:
It's necessary cause generic node-containers are always created for all k8s clusters - even for single pool clusters. There is a test case[0] that shows a DS created with a affinity that never gonna be evaluated on a cluster without Tsuru labels for nodes (eg. on EKS). For single pool cluster mode you should always create a specific entry for the related pool if you want it running - generic entry will not work. 
[0] https://github.com/tsuru/tsuru/pull/2411/files/03a21c146c9994ef0bc3eb2d4b5ea016875c9d5e#diff-9df82fdc12e61e08f6e990b6fa5be696R326

Reply:
Did you consider not creating the `-all` DS at all? We could do something like:
```
if singlePool {
	 if pool == "" {
		return nil
	}
	affinity = &apiv1.Affinity{}
}
```

I think this would make it easier to debug the cluster because we wouldn't have an extra DaemonSet the would never match any node.